hs_core/views/__init__.py

```diff
@@ -318,6 +347,21 @@ def set_resource_flag(request, shortkey, *args, **kwargs):
         _set_resource_sharing_status(request, user, res, flag_to_set='public', flag_value=True)
     elif t == 'make_private' or t == 'make_not_discoverable':
         _set_resource_sharing_status(request, user, res, flag_to_set='public', flag_value=False)
+
+       # reverse lookup: find all associated collection resources
+        associated_collection_metadata_obj_list = list(res.collection_set.all())
```

Comment:
corrected a typo in above comment: 
Another related but different scenario, if user delete ResB (this also happens on ResB's landing page), then CollectionA holds 0 resources which ## **does not** meet public criteria anymore, so we need to downgrade CollectionA to private.